enhancements/machine-api/aws-efa-networking.md

```diff
@@ -0,0 +1,253 @@
+today, as such, the EFA network interface will be the primary network interface for the instance.
+
+#### What is EFA?
+
+To quote the [AWS documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html):
+
+> An Elastic Fabric Adapter (EFA) is a network device that you can attach to your Amazon EC2 instance to accelerate
+  High Performance Computing (HPC) and machine learning applications. EFA enables you to achieve the application
+  performance of an on-premises HPC cluster, with the scalability, flexibility, and elasticity provided by the AWS
+  Cloud.
+>
+> EFA provides lower and more consistent latency and higher throughput than the TCP transport traditionally used in
+  cloud-based HPC systems. It enhances the performance of inter-instance communication that is critical for scaling HPC
+  and machine learning applications. It is optimized to work on the existing AWS network infrastructure and it can
+  scale depending on application requirements.
+
+### Risks and Mitigations
+
+#### This feature cannot be leveraged for control plane instances
+
+Due to limitations of the libraries that the installer uses to create control plane machines, there is currently no
+way in which we can enable the control plane machines to be created day 0 with the EFA network interface type.
+
+As this feature is only intended to be used with specific high performance workloads, which should not be running on
+control plane hosts, we do not expect end users to require EFA network interfaces for their control plane machines.
+
+#### Users will have to configure their own security groups
+
+To allow the feature to work, the security groups that contain the Machines must be configured to allow all traffic
+between the hosts within the security group. This is not configured by default on an IPI cluster.
+Users must manually make this change via the AWS console/cli.
+This prerequisite should be documented as a manual step to enable the feature.
+
+Additionally, as we have future plans to manage security groups for IPI clusters, these manual changes to the security
+groups may be difficult to adopt into the future management system. However, there is nothing to stop users of IPI
+clusters making changes such as this already, so this kind of change must be considered for any adoption plans already.
+
+Additionally, IPI users are free to add additional security groups to their clusters post install and specify these
+additional groups within their Machine specs. We have no control over this today and again, this is something that must
+be planned for in the future security group management process.
+
+#### Using EFA requires additional software
+
+The EFA network interface types require special drivers to enable them to be used.
+Fortunately, the RHEL kernel already includes the relevant kernel modules to allow the adapter to be used transparently
```

Comment:
state a goal up top of pinning this behavior forever.  Is there a way for you to ensure this is always the case or will the testing need to be manual?  You need the testing not just to scale the machine, but to make sure the EFA feature is useable.

Reply:
The kernel modules are available out of the box in both RHEL8 and RHEL9, there should be no risk of them being removed in the foreseeable future.

I'd prefer we explicitly state that the scope here is limited to bringing up instances with the EFA interface and ensuring they work in an existing cluster. Everything that happens via the third-party daemonset is explicitly out of scope.

Reply:
> I'd prefer we explicitly state that the scope here is limited to bringing up instances with the EFA interface and ensuring they work in an existing cluster. Everything that happens via the third-party daemonset is explicitly out of scope.

I don't think it's reasonable to say that we enable EFAs without knowing whether those EFAs we enabled can actually work. If we don't test this and it works in 4.10 and fails in 4.11, that would be a pretty serious regression.  Even if we don't attempt to fix the problem, knowing that it broke allows us to hold the edge on that configuration.

Reply:
@deads2k as I see it, the QE will test the actual consumption of the EFA interfaces. And again, I dont think we should support a flow where EFA interfaces is being used as standard interfaces - for that they CUs can use the default interface, right?

Reply:
RFE-2236 and OCPCLOUD-1353 make no mention of libfabric. That's out of scope, as would be any placement policies, at least until we see followup RFEs.

I agree scaling up nodes with EFA and the smoke test is sufficient to meet the requirements as they stand today.

Reply:
Lets skip to the scope of OCPCLOUD-1353  - to enhance MAPI to enable EFA instance types.  A reasonable CI test for this can be what Joel suggests above. "A simpler test that we can do is just to create a machineset with the feature enabled and scale it up, check that the node joins the cluster, then scale down again."